cmd/crossplane/core/core.go

```diff
@@ -204,24 +206,34 @@ func (c *startCommand) Run(s *runtime.Scheme, log logging.Logger) error { //noli
 			log.Info("Beta feature enabled", "flag", features.EnableBetaCompositionFunctionsExtraResources)
 		}
 
-		clienttls, err := certificates.LoadMTLSConfig(
-			filepath.Join(c.TLSClientCertsDir, initializer.SecretKeyCACert),
-			filepath.Join(c.TLSClientCertsDir, corev1.TLSCertKey),
-			filepath.Join(c.TLSClientCertsDir, corev1.TLSPrivateKeyKey),
-			false)
-		if err != nil {
-			return errors.Wrap(err, "cannot load client TLS certificates")
-		}
-
 		m := xfn.NewMetrics()
 		metrics.Registry.MustRegister(m)
 
-		// We want all XR controllers to share the same gRPC clients.
-		functionRunner = xfn.NewPackagedFunctionRunner(mgr.GetClient(),
+		fnRunnerOpts := []xfn.PackagedFunctionRunnerOption{
 			xfn.WithLogger(log),
-			xfn.WithTLSConfig(clienttls),
 			xfn.WithInterceptorCreators(m),
-		)
+		}
+
+		if c.TLSAllowInsecure {
+			fnRunnerOpts = append(fnRunnerOpts, xfn.WithTLSConfig(&tls.Config{
+				MinVersion:         tls.VersionTLS12,
+				InsecureSkipVerify: true, //nolint:gosec // User setting
+			}))
```

Comment:
Does this definitely work as expected?

When a function runs with `--insecure` it uses gRPC's `insecure.NewCredentials`. I thought this completely disabled TLS, causing the function to serve plaintext. Is that not the case?

Reply:
Could we keep it simple and add just an `--insecure` flag (to Crossplane) that causes it to:

* Connect to functions using plaintext.
* Connect to webhooks (and external secret stores?) as plaintext?

I like that:

* This matches the existing `--insecure` flag at the function end.
* (iirc) this matches the `--insecure` flag that various Kubernetes components use

I'd prefer to avoid the complexity of having two flags - one to use plaintext, and another to skip TLS certificate verification - if the goal is just to enable local debugging without dealing with TLS.